### ISSUE TITLE
UnpicklingError when using pyenv global python

### ISSUE THREAD

**Issue by sloria** [davidhalter/jedi#1271]
## Versions

Python and jedi

```
Python 3.7.2 (default, Jan 14 2019, 09:51:27)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import jedi

In [2]: jedi.__version__
Out[2]: '0.12.1'
```

macOS 10.14.2 (Mojave)

MacVim

```
:version
VIM - Vi IMproved 8.1 (2018 May 18, compiled Dec 13 2018 12:07:57)
macOS version
Included patches: 1-577
Compiled by Homebrew
Huge version with MacVim GUI.  Features included (+) or not (-):
+acl               +cmdline_compl     +emacs_tags        +iconv             +mksession         -mzscheme          +reltime           +termguicolors     +visualextra
+arabic            +cmdline_hist      +eval              +insert_expand     +modify_fname      +netbeans_intg     +rightleft         +terminal          +viminfo
+autocmd           +cmdline_info      +ex_extra          +job               +mouse             +num64             +ruby              +terminfo          +vreplace
+autochdir         +comments          +extra_search      +jumplist          +mouseshape        +odbeditor         +scrollbind        +termresponse      +wildignore
-autoservername    +conceal           +farsi             +keymap            +mouse_dec         +packages          +signs             +textobjects       +wildmenu
+balloon_eval      +cryptv            +file_in_path      +lambda            -mouse_gpm         +path_extra        +smartindent       +timers            +windows
+balloon_eval_term +cscope            +find_in_path      +langmap           -mouse_jsbterm     +perl              +startuptime       +title             +writebackup
+browse            +cursorbind        +float             +libcall           +mouse_netterm     +persistent_undo   +statusline        +toolbar           -X11
++builtin_terms    +cursorshape       +folding           +linebreak         +mouse_sgr         +postscript        -sun_workshop      +transparency      -xfontset
+byte_offset       +dialog_con_gui    -footer            +lispindent        -mouse_sysmouse    +printer           +syntax            +user_commands     +xim
+channel           +diff              +fork()            +listcmds          +mouse_urxvt       +profile           +tag_binary        +vartabs           -xpm
+cindent           +digraphs          +fullscreen        +localmap          +mouse_xterm       -python            +tag_old_static    +vertsplit         -xsmp
+clientserver      +dnd               -gettext           +lua               +multi_byte        +python3           -tag_any_white     +virtualedit       -xterm_clipboard
+clipboard         -ebcdic            -hangul_input      +menu              +multi_lang        +quickfix          -tcl               +visual            -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/Applications/MacVim.app/Contents/Resources/vim"
Compilation:
clang -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_MACVIM -Wall -Wno-unknown-pragmas -pipe  -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -L.  -L/usr/local/lib -o Vim -framework Cocoa -framework Carbon       -lm  -lncurses -liconv -framework AppKit  -L/usr/local/opt/lua/lib -llua5.3 -fstack-protec
tor  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -lperl  -L/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/config-3.7m-darwin -lpython3.
7m -framework CoreFoundation  -framework Ruby
```

pyenv

```
❯ pyenv --version
pyenv 1.2.9
```

## Steps

1. `pyenv install 3.7.0`
2. Make 3.7.0 the default global python: `echo "3.7.0" > ~/.pyenv/version`
3. Open vim
4. Run `jedi#goto` on anything

## Error

```
Traceback (most recent call last):
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/cache.py", line 141, in wrapper
    return dct[key]
KeyError: ((), frozenset())
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi_vim.py", line 140, in wrapper
    return func(*args, **kwargs)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi_vim.py", line 302, in goto
    definitions = script.goto_assignments(follow_imports=True)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/api/__init__.py", line 262, in goto_assignments
    context = self._evaluator.create_context(self._get_module(), tree_name)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/api/__init__.py", line 148, in _get_module
    import_names = dotted_path_in_sys_path(self._evaluator.get_sys_path(), self.path)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/evaluate/__init__.py", line 126, in get_sys_path
    return self.project._get_sys_path(self, environment=self.environment)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/evaluate/cache.py", line 40, in wrapper
    rv = function(obj, *args, **kwargs)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/api/project.py", line 104, in _get_sys_path
    sys_path = list(self._get_base_sys_path(environment))
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/api/project.py", line 88, in _get_base_sys_path
    sys_path = list(environment.get_sys_path())
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/cache.py", line 143, in wrapper
    result = method(self, *args, **kwargs)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/api/environment.py", line 128, in get_sys_path
    return self._get_subprocess().get_sys_path()
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/evaluate/compiled/subprocess/__init__.py", line 202, in get_sys_path
    return self._send(None, functions.get_sys_path, (), {})
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/evaluate/compiled/subprocess/__init__.py", line 242, in _send
    is_exception, traceback, result = pickle_load(self._process.stdout)
  File "/Users/sloria/.vim/plugged/jedi-vim/pythonx/jedi/jedi/_compatibility.py", line 464, in pickle_load
    return pickle.load(file, encoding='bytes')
_pickle.UnpicklingError: invalid load key, '\x1b'.
```

## Notes

* This does not happen if using a pyenv virtualenv. It only happens when using the global python
* It doesn't seem to matter what is in `~/.pyenv/version`. I've tried using `system` as the first entry, but that didn't help
* Seems related to #1239.

**Comment by davidhalter** [comment]
Can you retry with latest Jedi? 0.12.1 is quite old. We might have fixed this?!

**Comment by sloria** [comment]
Sorry, I'm not using jedi-vim (using jedi through python-language-server) at the moment and won't have time to test this myself.

**Comment by davidhalter** [comment]
Fair enough. I'll close it in that case. 0.12 will definitely not work. We have fixed quite a few (similar) issues in 0.13.x. Once you have an issue again with 0.13.3+, please let me know! (You can generally just upgrade Jedi in python-language-server and check if it's better, the API changes between versions are usually minimal).
